moveit_kinematics/ikfast_kinematics_plugin/templates/ikfast61_moveit_plugin_template.cpp

```diff
@@ -1060,13 +1093,41 @@ bool IKFastKinematicsPlugin::getPositionIK(const geometry_msgs::Pose& ik_pose, c
     ROS_DEBUG_STREAM_NAMED("ikfast", "No IK solution");
   }
 
+  // Sort the solutions under limits and find the one that is closest to ik_seed_state
```

Comment:
I believe this isn't even correct bubble sort: The `swap_flag` is never read. You could add a `while (swap_flag)` around the whole thing, making it O(n^3). Alternatively, Wikipedia has an O(n^2) implementation of bubble sort. But I think I like the std::sort suggestion best. *g*